api/types.proto

```diff
@@ -713,3 +713,33 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+	NODE = 1 [(gogoproto.enumvalue_customname) = "NodeSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this reference.
+	string name = 1;
```

Comment:
I'm not sure.

We have a convention of having an ID in every top-level object, and we probably don't want to break that convention.

But the reason I raised the question is that `SecretDataID` below references an ID. Something feels a little strange about needing both a name and an opaque ID to reference a secret. At least using the ID of the `Secret` and the ID of the `SecretData` would be a little more consistent. But maybe there's a good reason to use a name for the `Secret`; it's hard for me to tell without seeing the rest of the code.